app/src/main/java/com/appdev/eateryblueandroid/ui/viewmodels/SearchViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.appdev.eateryblueandroid.ui.viewmodels
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.appdev.eateryblueandroid.models.Eatery
+import com.appdev.eateryblueandroid.networking.ApiService
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+class SearchViewModel(
+    fetchFromApi: Boolean
+): ViewModel() {
+    sealed class State{
+        object Loading : State()
+//        data class WordsTyped(val eateries: List<Eatery>): State()
+        object NothingTyped: State()
+//        object WordsTyped: State()
+        data class WordsTyped(val eateries: List<Eatery>): State()
+        object SearchResults: State()
+
+        data class Failure(val errorMsg: String) : State()
```

Comment:
I think states for this viewmodel should be Searching(val input: String): State() and Failure(val errorMsg: String): State. Should be getting the list of eateries from the HomeViewModel which we pass into the searchscreen, and just filter in the searchscreen based on those eateries and input string

Reply:
Agreed, I think we need a repo where the data is stored and then we can avoid an API call.